plugins/janus_streaming.c

```diff
@@ -2155,6 +2195,115 @@ struct janus_plugin_result *janus_streaming_handle_message(janus_plugin_session
+			g_snprintf(error_cause, 512, "No configuration file, can't edit mountpoint permanently");
+			goto plugin_response;
+		}
+		guint64 id_value = json_integer_value(id);
+		janus_mutex_lock(&mountpoints_mutex);
+		janus_streaming_mountpoint *mp = g_hash_table_lookup(mountpoints, &id_value);
+		if(mp == NULL) {
+			janus_mutex_unlock(&mountpoints_mutex);
+			JANUS_LOG(LOG_ERR, "No such mountpoint (%"SCNu64")\n", mp->id);
+			error_code = JANUS_STREAMING_ERROR_NO_SUCH_MOUNTPOINT;
+			g_snprintf(error_cause, 512, "No such mountpoint (%"SCNu64")", mp->id);
+			goto plugin_response;
+		}
+		janus_mutex_lock(&mp->mutex);
+		/* A secret may be required for this action */
+		JANUS_CHECK_SECRET(mp->secret, root, "secret", error_code, error_cause,
+			JANUS_STREAMING_ERROR_MISSING_ELEMENT, JANUS_STREAMING_ERROR_INVALID_ELEMENT, JANUS_STREAMING_ERROR_UNAUTHORIZED);
+		if(error_code != 0) {
+			janus_mutex_unlock(&mp->mutex);
+			janus_mutex_unlock(&mountpoints_mutex);
+			goto plugin_response;
+		}
+		/* Edit the mountpoint properties that were provided */
+		if(desc != NULL && strlen(json_string_value(desc)) > 0) {
+			char *old_description = mp->description;
+			char *new_description = g_strdup(json_string_value(desc));
+			mp->description = new_description;
+			g_free(old_description);
+		}
+		if(is_private)
+			mp->is_private = json_is_true(is_private);
+		/* A secret may be required for this action */
+		JANUS_CHECK_SECRET(mp->secret, root, "secret", error_code, error_cause,
+			JANUS_STREAMING_ERROR_MISSING_ELEMENT, JANUS_STREAMING_ERROR_INVALID_ELEMENT, JANUS_STREAMING_ERROR_UNAUTHORIZED);
+		if(error_code != 0) {
+			janus_mutex_unlock(&mountpoints_mutex);
+			goto plugin_response;
+		}
+		if(secret && strlen(json_string_value(secret)) > 0) {
+			char *old_secret = mp->secret;
+			char *new_secret = g_strdup(json_string_value(secret));
+			mp->secret = new_secret;
+			g_free(old_secret);
+		}
+		if(pin && strlen(json_string_value(pin)) > 0) {
+			char *old_pin = mp->pin;
+			char *new_pin = g_strdup(json_string_value(pin));
+			mp->pin = new_pin;
+			g_free(old_pin);
+		}
+		if(save) {
+			/* This change is permanent: save to the configuration file too
+			 * FIXME: We should check if anything fails... */
+			JANUS_LOG(LOG_VERB, "Modifying streaming mountpoint %"SCNu64" permanently in config file\n", mp->id);
+			janus_mutex_lock(&config_mutex);
+			char cat[BUFSIZ];
+			/* The mountpoint ID is the category */
+			g_snprintf(cat, BUFSIZ, "%"SCNu64, mp->id);
+			/* Remove the old category first */
+			janus_config_remove_category(config, cat);
+			/* Now write the mountpoint details again */
+			janus_config_add_category(config, cat);
+			janus_config_add_item(config, cat, "description", mp->description);
+			if(mp->is_private)
+				janus_config_add_item(config, cat, "is_private", "yes");
+			if(mp->secret)
+				janus_config_add_item(config, cat, "secret", mp->secret);
+			if(mp->pin)
+				janus_config_add_item(config, cat, "pin", mp->pin);
```

Comment:
This is broken too: you're removing the category with `janus_config_remove_category`, but only adding a small set of properties when writing it again (description, is_private, secret, pin), leaving out all the ton of properties the different mountpoints can have (see what the "create" code does for instance).

Reply:
Ah, thanks, will patch.